core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java

```diff
@@ -167,12 +170,27 @@ public void sendAdmin(Match match, MatchPlayer sender, @Nullable @Text String me
       usage = "[player] [message]")
   public void sendDirect(Match match, MatchPlayer sender, Player receiver, @Text String message) {
     if (sender == null) return;
+
+    if (vanish.isVanished(sender.getId())) {
+      sender.sendWarning(new PersonalizedTranslatable("vanish.chat.deny"));
+      return;
+    }
     if (isMuted(sender)) {
       sendMutedMessage(sender);
       return;
     }
     MatchPlayer matchReceiver = manager.getPlayer(receiver);
     if (matchReceiver != null) {
+
+      // Vanish Check - Don't allow messages to vanished
+      if (vanish.isVanished(matchReceiver.getId())) {
+        sender.sendMessage(
+            new PersonalizedTranslatable("command.playerNotFound")
+                .getPersonalizedText()
+                .color(ChatColor.RED));
+        return;
+      }
```

Comment:
Exclude staff from this? As in staff can send vanished staff messages? Maybe its enough with adminchat, what do you think?

Reply:
Will there be any reason to message other staff members? Using `/a` should suffice in my opinion and most staff usually know each other outside of the server, so any private conversations can be held via other platforms.